napari/components/overlays/labels_polygon.py

```diff
@@ -0,0 +1,43 @@
+from napari.components.overlays.base import SceneOverlay
+from napari.layers import Labels
+
+
+class LabelsPolygonOverlay(SceneOverlay):
+    """Overlay that displays a polygon on a scene.
+
+        It handles the following mouse events to update the overlay:
+        - Mouse move: Continuously redraw the latest polygon point with the current mouse position.
+        - Mouse press (left button): Adds the current mouse position as a new polygon point.
+        - Mouse double click (left button): If there are at least three points in the polygon and
+                      the double-click position is within completion_radius from the first vertex,
+                      the polygon will be painted in the image using the current label.
+
+        - Mouse press (right button): Removes the most recent polygon point from the list.
+
+    Attributes
+    ----------
+    enabled : bool
+        Controls whether the overlay is activated.
+    points : list
+        A list of (x, y) coordinates of the vertices of the polygon.
+    color : tuple
+        A tuple representing the RGBA color of the polygon.
+        Opacity only applies to the fill color of the polygon.
+        Borders have the same color, but they are always opaque.
+    double_click_completion : bool
+        Whether drawing the polygon can be completed with a left mouse double-click.
+    completion_radius : int
+        Defines the radius from the first polygon vertex within which
+        the drawing process can be completed by a left double-click.
+    """
+
+    enabled: bool = False
+    points: list = []
+    color: tuple = (1, 1, 1, 0.3)
```

Comment:
The polygon overlay should follow the opacity of the Labels layer, in which I set it to 0.5 at the beginning of the test: https://github.com/napari/napari/blob/15e37b6fb27a1d04066486cd01641a312b4b8641/napari/_vispy/_tests/test_vispy_labels_polygon_overlay.py#L18

Reply:
Actually, the polygon overlay do not need public `color` property as it only adds confusion with little utility because the polygon always follows the Labels current color. I removed it in this commit: https://github.com/napari/napari/pull/5806/commits/64ad4e7cb839f2eb0290349b1cec7827798276e8